iree/compiler/Conversion/LinalgToLLVM/LinalgBufferizePass.cpp

```diff
@@ -0,0 +1,447 @@
+
+// TODO: evolve into flow.interface.store.tensor when the abstraction to get
+// a raw void* / memref<?xi8> + offset + shape exists.
+// Then we can immediately emit a view op.
+// TODO: canonicalizations of the view op.
+LogicalResult convertInterfaceStoreTensorOp(
+    OpBuilder &b, IREE::HAL::InterfaceStoreTensorOp storeOp,
+    BlockAndValueMapping &bvm) {
+  OpBuilder::InsertionGuard g(b);
+  b.setInsertionPoint(storeOp);
+
+  // TODO: view + subview
+  if (!matchPattern(storeOp.offset(), m_Zero())) {
+    return storeOp.emitError("unhandled non-zero offset");
+  }
+
+  // Potentially duplicate placeholders will be taken care of by CSE.
+  auto phOp = createPlaceholderOp(b, storeOp.getLoc(), storeOp,
+                                  storeOp.operand(), storeOp.queryBindingOp());
+  Value buffer = phOp.getResult();
+  b.create<linalg::CopyOp>(storeOp->getLoc(), bvm.lookup(storeOp.operand()),
+                           buffer);
+  storeOp->erase();
+  return success();
+  ;
+}
+
+// TODO: evolve into flow.interface.store.tensor.tile when the abstraction
+// exists. This will turn into some subview chain from the base view.
+LogicalResult convertInterfaceStoreTensorOp(
+    OpBuilder &b, IREE::HAL::InterfaceStoreTensorTileOp storeOp,
+    BlockAndValueMapping &bvm) {
+  OpBuilder::InsertionGuard g(b);
+  b.setInsertionPoint(storeOp);
+
+  // TODO: view + subview
+  if (!matchPattern(storeOp.base_offset(), m_Zero())) {
+    return storeOp.emitError("unhandled non-zero offset");
+  }
+
+  // Potentially duplicate placeholders will be taken care of by CSE.
+  auto phOp =
+      createPlaceholderOp(b, storeOp.getLoc(), storeOp, storeOp.operand(),
+                          storeOp.queryBindingOp(), /*typeErase=*/true);
+  Value buffer = phOp.getResult();
+  auto inferredType = SubViewOp::inferResultType(
+      buffer.getType().cast<MemRefType>(),
+      extractFromI64ArrayAttr(storeOp.static_offsets()),
+      extractFromI64ArrayAttr(storeOp.static_sizes()),
+      extractFromI64ArrayAttr(storeOp.static_strides()));
+  Value subview = b.create<SubViewOp>(
+      storeOp->getLoc(), inferredType, buffer, storeOp.offsets(),
+      storeOp.sizes(), storeOp.strides(), storeOp.static_offsets(),
+      storeOp.static_sizes(), storeOp.static_strides());
+  b.create<linalg::CopyOp>(storeOp->getLoc(), bvm.lookup(storeOp.operand()),
+                           subview);
+  storeOp->erase();
+  return success();
+}
+
+namespace {
+struct LinalgLLVMBufferizePass
+    : public PassWrapper<LinalgLLVMBufferizePass, FunctionPass> {
+  void getDependentDialects(DialectRegistry &registry) const override {
+    registry.insert<IREEDialect, linalg::LinalgDialect, scf::SCFDialect,
+                    StandardOpsDialect>();
+  }
+  void runOnFunction() override;
+};
+}  // namespace
+
+void LinalgLLVMBufferizePass::runOnFunction() {
+  FuncOp funcOp = getFunction();
+
+  // Bail on any control-flow for now.
+  bool fail = false;
+  funcOp.walk([&](Operation *op) {
```

Comment:
Could just make this check for a single basic block?

Reply:
Actually I can remove that completely since inplace semantics are not taken advantage of in this pass.